spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  describe '#authenticate_admin' do
```

Comment:
Testing a private method is a bit of a code smell. The commit for this could use some more words, too!